api/oc_collection.c

```diff
@@ -807,18 +800,10 @@ oc_handle_collection_linked_list_request(oc_request_t *request)
       oc_endpoint_t *eps =
         oc_connectivity_get_endpoints(link->resource->device);
       while (eps != NULL) {
-        /* If this resource has been explicitly tagged as SECURE on the
-         * application layer, skip all coap:// endpoints, and only include
-         * coaps:// endpoints.
-         * Also, exclude all endpoints that are not associated with the
-         * interface through which this request arrived. This is achieved by
-         * checking if the interface index matches.
-         */
-        if (((link->resource->properties & OC_SECURE) &&
-             !(eps->flags & SECURED)) ||
-            (request->origin && request->origin->interface_index != -1 &&
-             request->origin->interface_index != eps->interface_index)) {
-          goto next_eps;
+        if (oc_filter_out_ep_for_resource(eps, link->resource, request->origin,
```

Comment:
The `oc_filter_out_ep_for_resource` function features this additional code

```C
  if (request_origin &&
      (((request_origin->flags & IPV4) && (ep->flags & IPV6)) ||
       ((request_origin->flags & IPV6) && (ep->flags & IPV4)))) {
    return true;
  }
```

which is not in the original code. Is that OK?

Reply:
It is :) it's from discovery but not used and we want to use it :)